packages/miniflare/src/plugins/core/index.ts

```diff
@@ -374,14 +375,39 @@ export const CORE_PLUGIN: Plugin<
     sourceMapRegistry,
   }) {
     // Define regular user worker
+    const additionalModuleNames = additionalModules.map(({ name }) => name);
     const workerScript = getWorkerScript(
       sourceMapRegistry,
       options,
-      workerIndex
+      workerIndex,
+      additionalModuleNames
     );
     // Add additional modules (e.g. "__STATIC_CONTENT_MANIFEST") if any
     if ("modules" in workerScript) {
-      workerScript.modules.push(...additionalModules);
+      const subDirs = new Set(
+        workerScript.modules.map(({ name }) => path.posix.dirname(name))
+      );
+      // Ignore `.` as it's not a subdirectory, and we don't want to register
+      // additional modules in the root twice.
+      subDirs.delete(".");
+
+      for (const module of additionalModules) {
+        workerScript.modules.push(module);
+        // In addition to adding the module, we add stub modules in each
+        // subdirectory re-exporting each additional module. These allow
+        // additional modules to be imported in every directory.
+        for (const subDir of subDirs) {
+          const relativePath = path.posix.relative(subDir, module.name);
+          const relativePathString = JSON.stringify(relativePath);
+          workerScript.modules.push({
+            name: path.posix.join(subDir, module.name),
+            // TODO(someday): if we ever have additional modules without
+            //  default exports, this may be a problem, for now, our only
+            //  additional module is `__STATIC_CONTENT_MANIFEST` so this is fine
```

Comment:
I guess we could read the file and look for occurrences of either `/\bexport\s+default\b/` or `\bas\s+default\b` and use that to determine whether we include the second ` export { default } from` line.